examples/simulation/train.ipynb

```diff
@@ -0,0 +1,289 @@
+    "import os\n",
+    "RunningInCOLAB = 'google.colab' in str(get_ipython())\n",
+    "if RunningInCOLAB:\n",
+    "    !wget https://raw.githubusercontent.com/lyft/l5kit/master/examples/setup_notebook_colab.sh -q\n",
+    "    !sh ./setup_notebook_colab.sh\n",
+    "    os.environ[\"L5KIT_DATA_FOLDER\"] = open(\"./dataset_dir.txt\", \"r\").read().strip()\n",
+    "else:\n",
+    "    print(\"Not running in Google Colab.\")\n",
+    "    os.environ[\"L5KIT_DATA_FOLDER\"] = \"/tmp/l5kit_data\""
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "dm = LocalDataManager(None)\n",
+    "# get config\n",
+    "cfg = load_config_data(\"./config.yaml\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# rasterisation\n",
+    "rasterizer = build_rasterizer(cfg, dm)\n",
+    "\n",
+    "# ===== INIT DATASET\n",
+    "train_zarr = ChunkedDataset(dm.require(cfg[\"train_data_loader\"][\"key\"])).open()\n",
+    "train_dataset = AgentDataset(cfg, train_zarr, rasterizer)\n",
+    "\n",
+    "# plot some examples\n",
+    "for idx in range(0, len(train_dataset), len(train_dataset) // 10):\n",
+    "    data = train_dataset[idx]\n",
+    "    im = rasterizer.to_rgb(data[\"image\"].transpose(1, 2, 0))\n",
+    "    target_positions = transform_points(data[\"target_positions\"], data[\"raster_from_agent\"])\n",
+    "    draw_trajectory(im, target_positions, TARGET_POINTS_COLOR)\n",
+    "    plt.imshow(im)\n",
+    "    plt.axis('off')\n",
+    "    plt.show()\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "model = PlanningModel(\n",
+    "        model_arch=\"resnet50\",\n",
+    "        num_input_channels=rasterizer.num_channels(),\n",
+    "        num_targets=3 * cfg[\"model_params\"][\"future_num_frames\"],  # X, Y, Yaw * number of future states,\n",
+    "        weights_scaling= [1., 1., 1.],\n",
+    "        criterion=nn.MSELoss(reduction=\"none\")\n",
+    "        )\n",
+    "print(model)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Prepare for training\n",
+    "Our `AgentDataset` inherits from PyTorch `Dataset`; so we can use it inside a `Dataloader` to enable multi-processing."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "train_cfg = cfg[\"train_data_loader\"]\n",
+    "train_dataloader = DataLoader(train_dataset, shuffle=train_cfg[\"shuffle\"], batch_size=train_cfg[\"batch_size\"], \n",
+    "                             num_workers=train_cfg[\"num_workers\"])\n",
+    "device = torch.device(\"cuda:0\" if torch.cuda.is_available() else \"cpu\")\n",
+    "model = model.to(device)\n",
+    "optimizer = optim.Adam(model.parameters(), lr=1e-3)\n",
```

Comment:
lr dropping with increased epoch number? At least add a TODO

Reply:
add a note in the cell before training